ci_framework/roles/tempest/tasks/prepare_cloud.yml

```diff
@@ -0,0 +1,32 @@
+---
+- name: Grab Openstack networks information
+  openstack.cloud.networks_info:
+    cloud: "{{ cloud_name }}"
+    filters:
+      router:external: true
+  register: osp_net_info
+
+- name: Create a dummy public network if there isn't
+  block:
+  - name: Create public network
+    openstack.cloud.network:
+      cloud: "{{ cloud_name }}"
+      state: present
+      name: "{{ public_network_name }}"
+      external: true
+    register: dummy_network
+
+  - name: Create public subnet
```

Comment:
Creating networks Does not really belongs to the test runner preparation.
It is something administrator should do regardless.

Probably it should  something like  os_post_config  role until there is a better idea.